src/unlink.js

```diff
@@ -63,7 +69,7 @@ module.exports = function unlink(config, args) {
     }
   }
 
-  const assets = dependency.assets;
+  const assets = diff(dependency.assets, flatMap(allDependencies, d => d.assets));
```

Comment:
A good task for tomorrow I'll have to do is to write unit tests for `unlink` part (just as we did for `link`).